apidef/oas/operation_test.go

```diff
@@ -0,0 +1,66 @@
+package oas
+
+import (
+	"testing"
+
+	"github.com/TykTechnologies/tyk/apidef"
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestOAS_PathsAndOperations(t *testing.T) {
+	const operationId = "userGET"
+	const existingOperationId = "userPOST"
+
+	var oas OAS
+	oas.Paths = openapi3.Paths{
+		"/user": {
+			Get: &openapi3.Operation{
+				OperationID: operationId,
+			},
+		},
+	}
+
+	xTykAPIGateway := &XTykAPIGateway{
+		Middleware: &Middleware{
+			Operations: Operations{
+				operationId: {
+					Allow: &Allowance{
+						Enabled: true,
+					},
+				},
+			},
+		},
+	}
+
+	oas.SetTykExtension(xTykAPIGateway)
+
+	var ep apidef.ExtendedPathsSet
+	oas.extractPathsAndOperations(&ep)
+
+	var convertedOAS OAS
+	convertedOAS.Paths = openapi3.Paths{
+		"/user": {
+			Post: &openapi3.Operation{
+				OperationID: existingOperationId,
+			},
+		},
+	}
+	convertedOAS.SetTykExtension(&XTykAPIGateway{Middleware: &Middleware{Operations: Operations{}}})
+	convertedOAS.fillPathsAndOperations(ep)
+
+	assert.Equal(t, oas.getTykOperations(), convertedOAS.getTykOperations())
+
```

Comment:
I'd like to see a failing test case here, particularly a non-existant OperationID (e.g. what error is expected if i pass "xxx" which doesn't have a match in .Paths).

Reply:
I think it should be handled in the schema validation at the first place.

Reply:
There are still dark code paths we can follow here - duplicate operationIds for one, and we can test for that behavior (should the first operationId take effect, or the last one, or... what do the tests say). Currently the tests are only testing for the "success" case, but not for failure.